source/common/rate_limiter_impl.cc

```diff
@@ -6,6 +6,39 @@
 
 namespace Nighthawk {
 
+BurstingRateLimiter::BurstingRateLimiter(RateLimiterPtr&& rate_limiter, const uint64_t burst_size)
+    : rate_limiter_(std::move(rate_limiter)), burst_size_(burst_size) {
+  ASSERT(burst_size_ > 0);
+}
+
+bool BurstingRateLimiter::tryAcquireOne() {
+  if (releasing_) {
+    accumulated_--;
+    releasing_ = accumulated_ > 0;
+    return true;
+  } else {
+    // We are greedy here, which should help with timely releases of bursts when
+    // the system is lagging behind.
+    while (rate_limiter_->tryAcquireOne()) {
+      accumulated_++;
+      if ((accumulated_ % burst_size_) == 0) {
```

Comment:
The use of `%` here together with both ++ on release and acquire is clever; it took me a minute to grok though, could you add a comment or two?

Reply:
Done